backend/src/controller/handleError.ts

```diff
@@ -19,14 +18,19 @@ function handleChatError(
 	statusCode = 500
```

Comment:
Hmm true i think at one point we did use distinguish between blocked messages and error messages for styling, but i don't think we need this defence information here now.

Reply:
I suppose you mean `isError` is always true, when we've sent the response via `handleChatError`.

Anyway, luckily not much tracing to do, all you need to do is look at `processChatResponse` in _ChatBox.tsx_. Yes, when `isError` is true, the `defenceReport `is irrelevant. But the `defenceReport` is relevant if `isError` is false. And in the frontend, we don't know ahead of time whether sending a message will cause an error or not, therefore we still need `isError` and `defenceReport` in `ChatHttpResponse`.

Reply:
I'm just checking the removed code now

Reply:
Does this mean we _could_ send null / undefined for defenceReport when we have a chat error? If so, then `defenceReport` in `ChatHttpResponse` would need to be optional, which is maybe not great for type-safety. In that case, maybe we could split the response into two types: ChatHttpResponse and ChatHttpErrorResponse.

That might be one to think about for future improvements though!

Reply:
the `defenceReport` in the response is necessarily not null/undefined! It just might be "empty", that is it might have a null `blockedReason`, or an empty `triggeredDefences` list. Which means the response is bigger than it needs to be when defences are not active, but It's fine for now